app/utils/file/index.ts

```diff
@@ -365,3 +371,71 @@ export function getLocalFilePathFromFile(dir: string, serverUrl: string, file: F
 
     return undefined;
 }
+
+export async function extractFileInfos(files: Array<Asset | DocumentPickerResponse | PastedFile>) {
+    const out: ExtractedFileInfo[] = [];
+
+    await Promise.all(files.map(async (file) => {
+        if (!file) {
+            return;
+        }
+
+        const outFile = {
+            progress: 0,
+            localPath: file.uri,
+            clientId: generateId(),
+            loading: true,
+        } as unknown as ExtractedFileInfo;
```

Comment:
Because this object's type does not overlap enough to be directly mapped to ExtractedFileInfo.  We can use Partial<ExtractedFileInfo> but that would be a longer road.